experimental/Schemes/elliptic_surface.jl

```diff
@@ -1207,15 +1239,15 @@ function horizontal_decomposition(X::EllipticSurface, F::Vector{QQFieldElem})
   @vprint :EllipticSurface 2 "Computing basis representation of $(P0)\n"
   p0 = basis_representation(X, P0_div) # this could be done from theory alone
   F1 = F - p0  # should be contained in the QQ-trivial-lattice
-  F2 = F1
   if all(isone(denominator(i)) for i in F1)
     # no torsion
     P = P0
     P_div = P0_div
+    F2 = F1
   else
     found = false
     for (i,(T, tor)) in enumerate(tors)
-      d = F2 - _vec(tor)
+      d = F2-vec(tor)
```

Comment:
@simonbrandhorst : Do you remember what was going on here in this file?

Reply:
the changes to linear algebra required changes elsewhere as well

Reply:
vec for Nemo matrices was renamed _vec